hs_core/hydroshare/resource.py

```diff
@@ -701,7 +701,10 @@ def delete_resource(pk):
     Note:  Only HydroShare administrators will be able to delete formally published resour
     """
 
-    res = utils.get_resource_by_shortkey(pk)
+    res = utils.get_resource_by_shortkey(pk, False)
```

Comment:
The reason is that this delete_resource() function is called by tests teardown() and in some cases, the resource has already been deleted so in teardown() method the resource is not available for deletion any more. I think when a resource is being requested to be deleted and it does not exist any more, we don't have to raise an exception since the fact the resource does not exist has the same effect as the fact the resource is deleted. If you can think of a use case that an exception is needed, I can revert it back and make the calling function in tests to check whether resource exists or not before deletion, but I think it is better to do it in lower level unless there is a use case against that.